frontend/src/components/administration/administration.tsx

```diff
@@ -1,26 +1,52 @@
-import { ReactElement } from 'react';
+import { ReactElement, useEffect, useState, ChangeEvent } from 'react';
 
-import { User } from '@autoline/shared/common/types/types';
+import { InputField } from '@components/common/input-field/input-field';
 import { Title } from '@components/common/title/title';
 import { Header } from '@components/header/header';
 import Container from '@mui/material/Container';
-import LinearProgress from '@mui/material/LinearProgress';
-import { useGetUsersQuery } from '@store/queries/users';
 
-import { UsersList } from './users/users-list';
+import { UsersListContainer } from './users/users-list-container';
+
+const useDebounce = (value: string, delay: number): string => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+};
```

Comment:
This must be outside of the component file, inside some shared hooks directory preferably. Because this hook can be reused in multiple places

Reply:
Yes, sure. That will be more correct. Done.